custom/src/main/java/co/elastic/otel/ElasticLogRecordExporter.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.otel;
+
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.logs.data.LogRecordData;
+import io.opentelemetry.sdk.logs.export.LogRecordExporter;
+import java.util.Collection;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class ElasticLogRecordExporter implements LogRecordExporter {
+  private static final AtomicReference<ElasticLogRecordExporter> INSTANCE = new AtomicReference<>();
+
+  private volatile boolean sendingLogs = true;
+  private final LogRecordExporter delegate;
+
+  public static ElasticLogRecordExporter getInstance() {
+    return INSTANCE.get();
+  }
+
+  static ElasticLogRecordExporter createCustomInstance(LogRecordExporter exporter) {
+    INSTANCE.set(new ElasticLogRecordExporter(exporter));
+    return INSTANCE.get();
+  }
+
+  private ElasticLogRecordExporter(LogRecordExporter delegate) {
+    this.delegate = delegate;
+  }
+
+  public void setSendingLogs(boolean send) {
+    sendingLogs = send;
+  }
+
+  public boolean sendingLogs() {
+    return sendingLogs;
+  }
+
+  @Override
+  public CompletableResultCode export(Collection<LogRecordData> collection) {
+    if (sendingLogs) {
+      return delegate.export(collection);
+    } else {
+      return CompletableResultCode.ofFailure();
```

Comment:
Maybe I'm overthinking this, though if the export wasn't carried out due to our config flag, isn't that "as expected"? meaning that the operation was successful in a way? I'm asking because I can see a case where some exporter (the disk exporter is one for example) might decide to try again later if it receives an "export failed" result, and I'm not sure if we'd like that to happen for this case. I'm fine with leaving it like this btw, but it's something I'd like to mention in case it wasn't considered before.

Reply:
I really don't know, I was 50/50 on it myself. I did a https://github.com/search?q=repo%3Aopen-telemetry%2Fopentelemetry-java%20CompletableResultCode.ofFailure&type=code check and the SDK sends failure on shutdown, stopping is a reversible shutdown so I went with failure. I'm still 50/50, I'll ask on the otel Java channel

Reply:
Reply from Lauri
```
BatchSpanProcessor logs a debug message on failed export, on successful export it increments a metric. Based on that I'd say that your exporter should return success.
```
so given everyone seems to feel it's success, I'll flip to that

Reply:
It wasn't fully clear to me either so I'm glad we could get someone like Lauri to confirm! Thank you.

Reply:
applied in c7b825d